File: python-uv/src/beginner/q01.py
#
# 10進数、8進数、2進数、すべてで回文数となる数値のうち、
# 10進数の10以上で最小の数は？
#
from datetime import datetime
from itertools import count

begin = datetime.now().microsecond

for i in count(start=10):
  b02 = format(i, "b")
  o08 = format(i, "o")
  d10 = format(i, "d")
  x16 = format(i, "x")

  if b02 == b02[::-1] and o08 == o08[::-1] and d10 == d10[::-1]:
    print(f"a01. = {i}, {x16}")
    break

finish = datetime.now().microsecond
print(f"{finish - begin}ms")

File: script/java/app/src/main/java/org/example/puzzle/beginner/Q01.java
package org.example.puzzle.beginner;

import java.util.Objects;

public class Q01 {
  public static void main(String[] args) {
    final var begin = System.currentTimeMillis();

    for (int i = 10;; i++) {
      final var b02 = Integer.toString(i, 2);
      final var o08 = Integer.toString(i, 8);
      final var d10 = Integer.toString(i, 10);
      final var x16 = Integer.toString(i, 16);

      if (Objects.equals(b02, new StringBuilder(b02).reverse().toString())
          && Objects.equals(o08, new StringBuilder(o08).reverse().toString())
          && Objects.equals(d10, new StringBuilder(d10).reverse().toString())) {
        System.out.println("a01. = " + i + ", " + x16);
        break;
      }
    }

    final var finish = System.currentTimeMillis();
    System.out.println((finish - begin) + "ms");
  }
}
